src/container/srv_target.c

```diff
@@ -1232,8 +1260,7 @@ cont_child_destroy_one(void *vin)
 			D_GOTO(out_pool, rc = -DER_BUSY);
 		} /* else: resync should have completed, try again */
 
-		/* If it is the last user, ds_cont_child will be removed from hash & freed. */
-		cont_child_put(tls->dt_cont_cache, cont);
+		daos_lru_ref_wait_evict(tls->dt_cont_cache, &cont->sc_list);
```

Comment:
How can we prevent new coming IO during the wait? If cannot, more and more new IO request may block the destroy process for ever?

Reply:
After discussed with shilong, it seems work well.